pkg/controller/machinepool/nutanixactuator_test.go

```diff
@@ -0,0 +1,277 @@
+package machinepool
+
+import (
+	"encoding/json"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	machinev1 "github.com/openshift/api/machine/v1"
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hivev1nutanix "github.com/openshift/hive/apis/hive/v1/nutanix"
+)
+
+func getMasterMachineWithImage(t assert.TestingT) *machineapi.Machine {
+	masterMachine := testMachine("master0", "master")
+
+	uuid := "some-machine-image-uuid"
+	name := "nutanix-image-name"
+	nmpc := machinev1.NutanixMachineProviderConfig{
+		Image: machinev1.NutanixResourceIdentifier{
+			Type: machinev1.NutanixIdentifierName,
+			UUID: &uuid,
+			Name: &name,
+		},
+	}
+	providerSpecValue, err := json.Marshal(nmpc)
+	assert.NoError(t, err, "unexpected error creating NutanixActuator")
+
+	masterMachine.Spec.ProviderSpec.Value = &runtime.RawExtension{
+		Raw: providerSpecValue,
+	}
+
+	return masterMachine
+}
+
+func TestNewNutanixActuator(t *testing.T) {
+	actuator, err := NewNutanixActuator(getMasterMachineWithImage(t))
+	assert.NoError(t, err, "unexpected error creating NutanixActuator")
+	assert.NotNil(t, actuator, "expected a valid NutanixActuator instance")
+}
+
+func TestGenerateMachineSets(t *testing.T) {
```

Comment:
Could use some coverage for the `getNutanixDataDisks()` [stack](https://app.codecov.io/gh/openshift/hive/pull/2573#0949c8a1be2450d692cbe4ee24935dde-R175).